lib/ansible/modules/windows/win_rds_rap.ps1

```diff
@@ -0,0 +1,252 @@
+#!powershell
+
+# Copyright: (c) 2018, Kevin Subileau (@ksubileau)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Module Ansible.ModuleUtils.SID
+
+$ErrorActionPreference = "Stop"
+
+# List of authentication methods as string. Used for parameter validation and conversion to integer flag, so order is important!
+$computer_group_types = @("rdg_group", "ad_network_resource_group", "allow_any")
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$diff_mode = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$description = Get-AnsibleParam -obj $params -name "description" -type "str"
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "absent","present","enabled","disabled"
+$computer_group_type = Get-AnsibleParam -obj $params -name "computer_group_type" -type "str" -validateset $computer_group_types
+$computer_group = Get-AnsibleParam -obj $params -name "computer_group" -type "str" -failifempty ($computer_group_type -eq "ad_network_resource_group" -or $computer_group_type -eq "rdg_group")
+$user_groups = Get-AnsibleParam -obj $params -name "user_groups" -type "list"
+$allowed_ports = Get-AnsibleParam -obj $params -name "allowed_ports" -type "list"
+
+
+function Get-RAP([string] $name) {
+    $rap_path = "RDS:\GatewayServer\RAP\$name"
+    $rap = @{
+        Name = $name
+    }
+
+    # Fetch RAP properties
+    Get-ChildItem -Path "$rap_path" | foreach { $rap.Add($_.Name,$_.CurrentValue) }
+    # Convert boolean values
+    $rap.Enabled = $rap.Status -eq 1
+    $rap.Remove("Status")
+
+    # Convert multiple choices values
+    $rap.ComputerGroupType = $computer_group_types[$rap.ComputerGroupType]
+
+    # Convert allowed ports from string to list
+    if($rap.PortNumbers -eq '*') {
+        $rap.PortNumbers = @("any")
+    } else {
+        $rap.PortNumbers = @($rap.PortNumbers -split ',')
+    }
+
+    # Fetch RAP user groups
+    $rap.UserGroups = @(Get-ChildItem -Path "$rap_path\UserGroups" | Select -ExpandProperty Name)
+
+    return $rap
+}
+
+function Set-RAPPropertyValue {
+    [CmdletBinding(SupportsShouldProcess=$true)]
```

Comment:
We've gotten rid of this linting rule as Ansible modules do things slightly differently from normal PowerShell cmdlets. Better to get rid of this as it technically doesn't support this.

Reply:
That's mean it's better to use a custom argument (like `-CheckMode`) or the global variable `$check_mode` instead of implementing the standard `-WhatIf` argument, is that right ?
The latter has the advantage of being automatically propagated in the call stack.